packages/patternfly-4/react-table/src/components/Table/Table.md

```diff
@@ -1202,3 +1206,420 @@ EmptyStateTable = () => {
   );
 }
 ```
+
+```js title=Editable-Rows
+import React from 'react';
+import { TextInput } from '@patternfly/react-core';
+import {
+  Table,
+  TableHeader,
+  TableBody,
+  TableVariant,
+  getErrorTextByValidator,
+  cancelCellEdits,
+  validateCellEdits,
+  applyCellEdits
+} from '@patternfly/react-table';
+
+class EditableRowsTable extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      columns: [
+        'Text input col 1',
+        'Disabled text input col 2',
+        'Text input col 3',
+        'Text input col 4'
+      ],
+      actions: [
+        {
+          title: 'Some action',
+          onClick: (event, rowId, rowData, extra) => console.log('clicked on Some action, on row: ', rowId)
+        },
+        {
+          title: <a href="https://www.patternfly.org">Link action</a>
+        },
+        {
+          isSeparator: true
+        },
+        {
+          title: 'Third action',
+          onClick: (event, rowId, rowData, extra) => console.log('clicked on Third action, on row: ', rowId)
+        }
+      ],
+      rows: [
+        {
+          isEditable: true,
+          rowEditValidationRules: [
+            {
+              name: 'required',
+              validator: value => value.trim() !== '',
+              errorText: 'This field is required'
+            }
+          ],
+          cells: [
+            {
+              title: (value, rowIndex, cellIndex, props) => (
+                <React.Fragment>
+                  <div className="pf-c-inline-edit__value">{value}</div>
```

Comment:
Why are we doing this inline like this?  Shouldn't it be part of an inline edit component?

Reply:
It's because Table does not edit or peer into its cells at all, the way it is currently. And the functional rendering allows the user to control what the input/value fields look like and how they behave.

I do believe we can wrap the whole `React.Fragment` into a simple component, to serve as a 'default' behavior/look.

Reply:
Right, what @kmcfaul said :D - I created a default component that people can get up and running with, can be replaced with a custom component as users see fit. Definitely makes things cleaner.. let me know if you see any issues!

Reply:
I still am not understanding.  We want the consumers to add the pf classnames on there own?

Reply:
Having the user specify the className like this doesn't seem like something we want to promote.  How have we handled this before in the table?  I don't recall us having to specify classNames like this with the table for components that are released as part of the PF4 react library ... should this be an InlineEditValue component... or have another component wrapped around this where we set this styling by passing in a prop?